src/olmo_core/model_ladder2/transformer_model_configurator.py

```diff
@@ -0,0 +1,197 @@
+            pct_diff := (
+                math.fabs(model.num_non_embedding_params - size_spec.num_params)
+                / size_spec.num_params
+            )
+        ) > 0.05:
+            warn_once(
+                f"Configured model has {format_count(model.num_non_embedding_params)} (non-embedding) parameters, "
+                f"which differs from target of {size_spec} by ~{100 * pct_diff:.1f}%.",
+                UserWarning,
+            )
+        return model
+
+    def configure_rank_microbatch_size(
+        self,
+        *,
+        size_spec: str,
+        sequence_length: int,
+        device_type: str,
+    ) -> int:
+        # TODO: configure context-parallelism if needed.
+        device_type = device_type.lower()
+        assert "h100" in device_type or "b200" in device_type
+        assert sequence_length in {2048, 4096, 8192}
+        size_spec = TransformerSize(size_spec)
+
+        num_params = size_spec.num_params
+        mbz: int
+        if num_params <= 190e6:
+            mbz = 16 * 4096
+        elif num_params <= 370e6:
+            mbz = 12 * 4096
+        elif num_params <= 760e6:
+            mbz = 10 * 4096
+        elif num_params <= 1e9:
+            mbz = 8 * 4096
+        elif num_params <= 3e9:
+            mbz = 4 * 4096
+        elif num_params <= 7e9:
+            mbz = 2 * 4096
+        else:
+            mbz = 2 * 4096
+
+        if "b200" in device_type:
+            mbz = mbz * 2
+
+        return mbz
+
+    def configure_minimal_device_mesh_spec(
+        self,
+        *,
+        size_spec: str,
+        sequence_length: int,
+        device_type: str,
+    ) -> DeviceMeshSpec:
+        # TODO: configure context-parallelism if needed.
+        device_type = device_type.lower()
+        assert "h100" in device_type or "b200" in device_type
+        assert sequence_length in {2048, 4096, 8192}
+        size_spec = TransformerSize(size_spec)
+
+        num_params = size_spec.num_params
+        if num_params < 13e9:
+            return DeviceMeshSpec(world_size=8, dp_world_size=None)
+        else:
+            return DeviceMeshSpec(world_size=32, dp_world_size=None)
+
+    def build_train_module(
+        self,
+        *,
+        size_spec: str,
+        sequence_length: int,
+        rank_microbatch_size: int,
+        model_config: TransformerConfig,
+        optim_config: OptimConfig,
+        scheduler: Scheduler,
+        device_type: str,
+    ) -> TransformerTrainModule:
+        # TODO: configure context-parallelism if needed.
+        device_type = device_type.lower()
+        assert "h100" in device_type or "b200" in device_type
+        assert sequence_length in {2048, 4096, 8192}
+        size_spec = TransformerSize(size_spec)
+
+        dp_config = TransformerDataParallelConfig(
+            name=DataParallelType.fsdp,
+            param_dtype=DType.bfloat16,
+            reduce_dtype=DType.float32,
+            wrapping_strategy=TransformerDataParallelWrappingStrategy.blocks,
```

Comment:
nit: I think full wrapping is slightly better than blocks in general. This is because under blocks wrapping the LM head isnt released for communication until after the backward pass is complete.